Assets/Tests/Runtime/ConnectionManagementTests.cs

```diff
@@ -0,0 +1,593 @@
+            }
+
+            m_ServerConnectionStateSequence = new List<ConnectionState>();
+            InitializeInstance("Server", m_ServerNetworkManager, m_ServerConnectionStateSequence, playerPrefabGO, out m_ServerConnectionManager, out m_ServerScope);
+
+            base.OnServerAndClientsCreated();
+        }
+
+        protected override IEnumerator OnTearDown()
+        {
+            m_ServerConnectionManager.RequestShutdown();
+
+            yield return new WaitWhile(() => m_ServerNetworkManager.IsListening);
+
+            for (var i = 0; i < NumberOfClients; i++)
+            {
+                var clientId = i;
+                yield return new WaitWhile(() => m_ClientNetworkManagers[clientId].IsListening);
+            }
+
+            m_ServerScope.Dispose();
+
+            for (var i = 0; i < NumberOfClients; i++)
+            {
+                m_ClientScopes[i].Dispose();
+            }
+
+            foreach (var sceneGameObject in GameObject.FindObjectsOfType<GameObject>())
+            {
+                GameObject.DestroyImmediate(sceneGameObject);
+            }
+            yield return base.OnTearDown();
+        }
+
+        void StartHost()
+        {
+            m_ServerConnectionManager.StartHostIp("server", "127.0.0.1", 9998);
+        }
+
+        IEnumerator ConnectClients()
+        {
+            for (var i = 0; i < NumberOfClients; i++)
+            {
+                m_ClientConnectionManagers[i].StartClientIp($"client{i}", "127.0.0.1", 9998);
+            }
+
+            yield return WaitForClientsConnectedOrTimeOut(m_ClientNetworkManagers);
+        }
+
+        void SetUniqueProfilesForEachClient()
+        {
+            for (var i = 0; i < NumberOfClients; i++)
+            {
+                var profileManager = m_ClientScopes[i].Container.Resolve<ProfileManager>();
+                profileManager.Profile = $"Client{i}";
+            }
+        }
+
+        [Test]
+        public void StartHost_Success()
+        {
+            StartHost();
+            Assert.IsTrue(m_ServerNetworkManager.IsHost);
+            var expectedServerConnectionStateSequence = new List<ConnectionState>();
+            expectedServerConnectionStateSequence.Add(m_ServerConnectionManager.m_StartingHost);
+            expectedServerConnectionStateSequence.Add(m_ServerConnectionManager.m_Hosting);
+            Assert.AreEqual(expectedServerConnectionStateSequence, m_ServerConnectionStateSequence);
+        }
+
+        [UnityTest]
+        public IEnumerator StartHostAndConnectClients_Success()
+        {
+            StartHost();
+            Assert.IsTrue(m_ServerNetworkManager.IsHost);
+
+            SetUniqueProfilesForEachClient();
+
+            yield return ConnectClients();
+            for (var i = 0; i < NumberOfClients; i++)
+            {
+                Assert.IsTrue(m_ClientNetworkManagers[i].IsConnectedClient);
```

Comment:
What else happens after a connection is established? Can we test that more things happened?

Reply:
On the client's side, nothing that concerns the ConnectionManager that is not already tested by NGO. On the host, we could test the state of the SessionManager, but I'm not entirely convinced this is pertinent to what is being tested here?